docs/ADR-4-Era-selectors-for-CLI-commands.md

```diff
@@ -0,0 +1,97 @@
+# Status
+
+📜 Proposed 2023-09-04
+
+# Context
+
+The Cardano block chain can be hard forked through multiple eras throughout its existence.  CLI commands may behave differently between eras depending on what the current era is and what era the user runs the command in.
+
+The era to be used for a CLI command can be optional in which case the command may connect to the node to find out what the current era is and choose an era appropriately.
+
+For these cases, the question arises what the default era should be for the CLI command.
+
+Moreover there is the question of how the default era should change over time, particularly in relation to a hard fork: whether the default era should change in the release before or after the hard fork happens.
+
+There is even scope to adopt a different strategy between query and transaction commands or even between different transaction commands.
+
+For example we can create a transaction in two different ways using the CLI - using `transaction build` and `transaction build-raw` commands.  The `transaction build` command is "magical" in that it tries to be user friendly by inferring transaction parameters wherever it can so the user doesn't have to, while the `transaction build-raw` is for more experienced users because it provides no inference and leaves it up to the user to supply all the necessary transaction parameters.
+
+Would it make sense for "magical" commands like the `transaction build` command to try to be more helpful and use the current era as the default?  That would _seem_ to be in the spirit of the command.
+
+Historically we changed the default transactions era in the first release after the hard fork in order to give the tool/dapp developers the possibility to update the default transaction era in their code after the hard fork - otherwise their tools would stop working after the hard fork.
+
+Care needs to be taken to avoid making a CLI release that does not work by default by defaulting to using an era that is not yet active. So that's why we have historically only updated the default era in the release after the hard fork, not the release before.
```

Comment:
```suggestion
Care needs to be taken to avoid making a CLI release that does not work by defaulting to using an era that is not yet active. So that's why we have historically only updated the default era in the release after the hard fork, not the release before.
```

I get what you meant, but I think this can be simplified without losing semantics.